host/libhackrf/src/hackrf.h

```diff
@@ -43,88 +43,730 @@ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSI
+ * 
+ * ## version string
+ * 
+ * Version string identifies the firmware version on the board. It can be read with the function @ref hackrf_version_string_read
+ * 
+ * ## USB API version
+ * 
+ * USB API version identifies the USB API supported by the device's firmware. It is coded as a xx.xx 16-bit value, and can be read by @ref hackrf_usb_api_version_read
+ * 
+ * Example of reading firmware and USB API version (from [hackrf_info.c](https://github.com/greatscottgadgets/hackrf/blob/eff4a20022ca5d7f11405c3cdeea6c4195e347d0/host/hackrf-tools/src/hackrf_info.c#L157-L178)):
+ * 
+ *  ```c
+ *  result = hackrf_version_string_read(device, &version[0], 255);
+ *  if (result != HACKRF_SUCCESS) {
+ *      fprintf(stderr,
+ *          "hackrf_version_string_read() failed: %s (%d)\n",
+ *          hackrf_error_name(result),
+ *          result);
+ *      return EXIT_FAILURE;
+ *  }
+ *
+ *
+ *  result = hackrf_usb_api_version_read(device, &usb_version);
+ *  if (result != HACKRF_SUCCESS) {
+ *      fprintf(stderr,
+ *          "hackrf_usb_api_version_read() failed: %s (%d)\n",
+ *          hackrf_error_name(result),
+ *          result);
+ *      return EXIT_FAILURE;
+ *  }
+ *  printf("Firmware Version: %s (API:%x.%02x)\n",
+ *      version,
+ *      (usb_version >> 8) & 0xFF,
+ *      usb_version & 0xFF);
+ *  ```
+ * 
+ * 
+ * ## Partid serial number
+ * 
+ * "part ID number", part id of the MCU. Read via @ref hackrf_board_partid_serialno_read. See the documentation of the MCU for details!
+ * 
+ * ## board revision
+ * 
+ * Board revision identifies revision of the HackRF board inside a device. Read via @ref hackrf_board_rev_read and converted into a human-readable string via @ref hackrf_board_rev_name. See @ref hackrf_board_rev for possible values. When reading, the value should be initialized with @ref BOARD_REV_UNDETECTED
+ * 
+ * ## supported platform
+ * 
+ * Identifies the platform supported by the firmare of the HackRF device. Read via @ref hackrf_supported_platform_read. Returns a bitfield. Can identify bad firmware version on device.
+ * 
+ */
+
+/**
+ * @defgroup configuration Configuration of the RF hardware
+ * 
+ * @brief configuring gain, sample rate, filter bandwidth, etc.
+ * 
+ * # Amplifiers and gains
+ * 
+ * There are 5 different amplifiers in the HackRF One. Most of them have variable gain, but some of them can be either enabled / disabled. Please note that most of the gain settings are not precise, and they depend on the used frequency as well.
+ * 
+ * ![hackrf components](https://hackrf.readthedocs.io/en/latest/_images/block-diagram.png)
+ * (image taken from https://hackrf.readthedocs.io/en/latest/hardware_components.html)
```

Comment:
An empty line is needed here to stop text following the large image on a line.

```suggestion
 * ![hackrf components](https://hackrf.readthedocs.io/en/latest/_images/block-diagram.png)
 *
 * (image taken from https://hackrf.readthedocs.io/en/latest/hardware_components.html)
```